dask/config.py

```diff
@@ -268,12 +275,30 @@ def interpret_value(value: str) -> Any:
     try:
         return ast.literal_eval(value)
     except (SyntaxError, ValueError):
-        if value.lower() in ("none", "null"):
+        if hasattr(value, "lower") and value.lower() in ("none", "null"):
```

Comment:
How's this possible?

Reply:
When `value` itself is `None`, only discovered when thinking about `dask config set <<key>>` ie, not setting a value. Seemed reasonable at the time. Maybe we ought to require an input for value though, I have no strong preference.

Reply:
`dask config set <<key>>` should be disallowed. The syntax should be `dask config set <<key>>=None`.
Explicit is better than implicit...
Forcing explicit with click would also save you the effort of writing dedicating unit tests.

Reply:
Done https://github.com/dask/dask/pull/10930/commits/b8a568f9ec9daf6d483c21545dc13da630fffd44

Reply:
Could you do the same to `dask config get` please?

Reply:
and done https://github.com/dask/dask/pull/10930/commits/2c1e6be3d9d141d4f269319448c6413431aea451